src/legacy/server/config/get_unused_config_keys.ts

```diff
@@ -0,0 +1,82 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import { difference, get, set } from 'lodash';
+// @ts-ignore
+import { transformDeprecations } from './transform_deprecations';
+import { unset, getFlattenedObject } from '../../utils';
+// @ts-ignore
+import { getTransform } from '../../deprecation';
+// eslint-disable-next-line @kbn/eslint/no-restricted-paths
+import { hasConfigPathIntersection } from '../../../core/server/config/';
+// eslint-disable-next-line @kbn/eslint/no-restricted-paths
+import { LegacyPluginSpec } from '../../../core/server/legacy/plugins/find_legacy_plugin_specs';
+// eslint-disable-next-line @kbn/eslint/no-restricted-paths
+import { LegacyConfig } from '../../../core/server/legacy/config';
```

Comment:
Note: that's a lot of 'internal' core imports. However we already had some in `src/legacy/server/config/complete.js` before I moved the logic here, and I'm afraid this can't be avoided until NP migration is complete.

I'm still not sure about the location of this file. this could be either here in legacy, will lot of core import, or I could move it to core in the same folder as `src/core/server/legacy/config/ensure_valid_configuration.ts` and have legacy imports in core instead. Not sure what we prefer?